src/test/java/test/listeners/github2385/IssueTest.java

```diff
@@ -0,0 +1,74 @@
+package test.listeners.github2385;
+
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+public class IssueTest extends SimpleBaseTest {
+    @Test
+    public void testExtendClass() {
+		TestNG testNG = create(SonTestClass.class);
+		testNG.run();
+        assertTrue(TestListener.listenerExecuted);
```

Comment:
As `TestListener.listenerExecuted` is static, the JVM keeps the value between the tests.
That's why you should reset the value before and/or after each test.

Reply:
Add reset in after method